local-interchaintest/src/utils/manager.rs

```diff
@@ -0,0 +1,210 @@
+    gc.chains = chain_infos.clone();
+
+    let mut uploader = test_ctx.build_tx_upload_contracts();
+    uploader.with_chain_name(NEUTRON_CHAIN_NAME);
+
+    // Get all contract paths
+    let authorization_contract_path = format!("{artifacts_dir}/{AUTHORIZATION_NAME}.wasm");
+    let processor_contract_path = format!("{artifacts_dir}/{PROCESSOR_NAME}.wasm");
+    let base_account_contract_path = format!("{artifacts_dir}/{BASE_ACCOUNT_NAME}.wasm");
+    let splitter_contract_path = format!("{artifacts_dir}/{SPLITTER_NAME}.wasm");
+    let reverse_splitter_contract_path = format!("{artifacts_dir}/{REVERSE_SPLITTER_NAME}.wasm");
+    let forwarder_contract_path = format!("{artifacts_dir}/{FORWARDER_NAME}.wasm");
+    let registry_contract_path = format!("{artifacts_dir}/{REGISTRY_NAME}.wasm");
+
+    // Upload all contracts
+    uploader.send_single_contract(&authorization_contract_path)?;
+    uploader.send_single_contract(&processor_contract_path)?;
+    uploader.send_single_contract(&base_account_contract_path)?;
+    uploader.send_single_contract(&splitter_contract_path)?;
+    uploader.send_single_contract(&reverse_splitter_contract_path)?;
+    uploader.send_single_contract(&forwarder_contract_path)?;
+    uploader.send_single_contract(&registry_contract_path)?;
+
+    // Get all contracts code ids
+    let authorization_code_id = test_ctx
+        .get_contract()
+        .contract(AUTHORIZATION_NAME)
+        .get_cw()
+        .code_id
+        .unwrap();
+    let processor_code_id = test_ctx
+        .get_contract()
+        .contract(PROCESSOR_NAME)
+        .get_cw()
+        .code_id
+        .unwrap();
+    let base_account_code_id = test_ctx
+        .get_contract()
+        .contract(BASE_ACCOUNT_NAME)
+        .get_cw()
+        .code_id
+        .unwrap();
+    let splitter_code_id = test_ctx
+        .get_contract()
+        .contract(SPLITTER_NAME)
+        .get_cw()
+        .code_id
+        .unwrap();
+    let reverse_splitter_code_id = test_ctx
+        .get_contract()
+        .contract(REVERSE_SPLITTER_NAME)
+        .get_cw()
+        .code_id
+        .unwrap();
+    let forwarder_code_id = test_ctx
+        .get_contract()
+        .contract(FORWARDER_NAME)
+        .get_cw()
+        .code_id
+        .unwrap();
+    let registry_code_id = test_ctx
+        .get_contract()
+        .contract(REGISTRY_NAME)
+        .get_cw()
+        .code_id
+        .unwrap();
+
+    // Update config with all the code ids
+    let mut code_ids_map = HashMap::new();
+    code_ids_map.insert(AUTHORIZATION_NAME.to_string(), authorization_code_id);
+    code_ids_map.insert(PROCESSOR_NAME.to_string(), processor_code_id);
+    code_ids_map.insert(BASE_ACCOUNT_NAME.to_string(), base_account_code_id);
+    code_ids_map.insert(SPLITTER_NAME.to_string(), splitter_code_id);
+    code_ids_map.insert(REVERSE_SPLITTER_NAME.to_string(), reverse_splitter_code_id);
+    code_ids_map.insert(FORWARDER_NAME.to_string(), forwarder_code_id);
```

Comment:
Can we pass this as an argument? Like have a struct something like:
```rust
pub struct ManagerSetupInfo {
       pub chain: String, // chain name
       pub contract_paths: Vec<String>, // path of all contracts we want to upload
}
```
and pass an array of this to this function? This way we can it up according to what we need for each test.
Right now it will only work for token_swap but would be nice that we can get it right from the start so that it can be reused for everything, including adding other domains.

We can still hard code the registry here to Neutron because we always need that.

This way each test can initialize the manager and tell it all the contracts that need to be uploaded and where

Reply:
Will look into it in the multi-chain manager

Reply:
OK so for now (and my astroport e2e test I'll just hardcode the contracts here for now)